### PR TITLE
Enhancement: UI fixes for the Side Menu screens

### PR DESCRIPTION
**Pull request by csbduzi**
## Contents of the PR:

- Dynamic arrow (back) button that appears/disappears depending on the screen you're coming from
- Fix for the screens Settings and About Us to be adjusted for all device screen sizes 
- Navigation link for the CourseScheduleDetails and NearbyInterestDetails to the DoubleSearch 
- Having a uniform back arrow on all screens

### Github Isssues

Issue #199

### CODE REVIEW

**Comment by mohanadarafe** [comment]
I got this bug when going to the _About Us_ page

IMG_8034

File: routes/routeStack.js [hunk 1]
```diff
@@ -43,7 +43,6 @@ const screens = {
         screen: PreferenceMenu,
         navigationOptions: {
             headerShown: false,
-            ...TransitionPresets.ModalSlideFromBottomIOS
```
Comment (mohanadarafe) [inline routes/routeStack.js:46]: why did you remove this?

File: screens/DoubleSearch.js [hunk 2]
```diff
@@ -62,7 +62,19 @@ function DoubleSearch(props) {
...
     const goBack = () => {
+        if(CourseScheduleDetailsScreen === true){
+            props.navigation.goBack();
+            props.navigation.navigate("CourseScheduleDetails")
+        }
+        else if (NearbyInterestDetailsScreen === true){
+            props.navigation.goBack();
+            props.navigation.navigate("NearbyInterestDetails")
+        }
         props.navigation.goBack();
```
Comment (mohanadarafe) [inline screens/DoubleSearch.js:78]: If both are false, line 78 `props.navigation.goBack();` goes back where?

File: screens/Menu/AboutUs.js [hunk 3]
```diff
@@ -35,23 +37,28 @@ function AboutUs (props) {
             "Hussain Witwit - hussainwitwit"
         ]
     }];
+    const textParagraph = [{
```
Comment (mohanadarafe) [inline screens/Menu/AboutUs.js:40]: Why is this not simply a string?

File: routes/routeStack.js (hunk 1 above)
Comment (csbduzi) [reply routes/routeStack.js:46]: Good catch! It wasn't supposed to removed

File: screens/Menu/NearbyInterest.js [hunk 4]
```diff
@@ -19,14 +20,49 @@ import PropTypes from "prop-types";
...
+     * The asyncstorage getter that will let us grab the value coming from the bottomMenu component
+     * @param  {} =>{letname=awaitAsyncStorage.getItem("sideMenu)"
+     * @param  {} ;setFromScreen(name)
+     */
+    const getFromScreen = async () => {
+        let name = await AsyncStorage.getItem("sideMenu");
+        setFromScreen(name);
+    };
+
+    useEffect(() => {
```
Comment (mohanadarafe) [inline screens/Menu/NearbyInterest.js:35]: Can we avoid useEffects & AsyncStorage and use props.navigation here?

File: screens/DoubleSearch.js (hunk 2 above)
Comment (csbduzi) [reply screens/DoubleSearch.js:78]: If both are false, then it goes back to either mapView or MoreDetails (it goes back to the screens it originated from, thats within the same navigation stack)

Comment (mohanadarafe) [reply screens/DoubleSearch.js:78]: Wouldn't it be better to add this in an _else_ bracket in that case?

File: screens/Menu/NearbyInterest.js (hunk 4 above)
Comment (csbduzi) [reply screens/Menu/NearbyInterest.js:35]: The original plan was to absolutely not use the AsyncStorage and the useEffects, but my hand was force because with props only, without using useEffect, there was no way to re-rendered the screen.

@AlainJobs  and I had a hour long discussion over it, the complexity comes from the fact that we use the navigation between two screens that are not in the same stack, meaning that the states of the components are never re-rendered, and the states of the hooks are never refreshed. (You would come back to NearbyInterest in the state of the hook would be the same as the previous time you had access the screen)

It would of been possible to re-render the screens if we could of use the goBack() method to pop out the screen from the stacks, but the goBack() does not work in this situation because screens **can only go back** to screens that are within the same stack, in this they are not.

Thus, since we could not use props, the final alternative was to use useEffecs & AsyncStorage

File: screens/Menu/AboutUs.js (hunk 3 above)
Comment (csbduzi) [reply screens/Menu/AboutUs.js:40]: Since I'm using a sectionList we need to pass it an array, not just a string

**Comment by csbduzi** [comment]
> I got this bug when going to the _About Us_ page
> 
> IMG_8034

Was this fixed with npm install ?

**Comment by mohanadarafe** [comment]
> I got this bug when going to the _About Us_ page
> 
> IMG_8034

Solved doing `npm i`

File: screens/DoubleSearch.js (hunk 2 above)
Comment (csbduzi) [reply screens/DoubleSearch.js:78]: Will do!
